### ISSUE TITLE
Call to undefined function Piwik\posix_getgrgid() after "core:update", Filechecks.php on line 268

### ISSUE THREAD

**Issue by ryrun** [piwik/piwik#10721]
I was updating my piwik installation and i got a php fatal error. I used the commadline for the update (console core:update). Cant reproduce it, because the update itself was successful (Everything is already up to date.).

Commandline output:

```
    *** Update ***

    Database Upgrade Required

    Your Piwik database is out-of-date, and must be upgraded before you can continue.

    Piwik database will be upgraded from version 2.16.2 to the new version 2.16.5.

    The following dimensions will be updated: log_visit.visit_entry_idaction_url.


This is a major update! It will take longer than usual.

    *** Note: this is a Dry Run ***

    ALTER TABLE `log_visit` MODIFY COLUMN `visit_entry_idaction_url` INTEGER(11) UNSIGNED NULL  DEFAULT NULL;

    *** End of Dry Run ***

A database upgrade is required. Execute update? (y/N) y

Starting the database upgrade process now. This may take a while, so please be patient.

    *** Update ***

    Database Upgrade Required

    Your Piwik database is out-of-date, and must be upgraded before you can continue.

    Piwik database will be upgraded from version 2.16.2 to the new version 2.16.5.

    The following dimensions will be updated: log_visit.visit_entry_idaction_url.

    The database upgrade process may take a while, so please be patient.

  Executing ALTER TABLE `log_visit` MODIFY COLUMN `visit_entry_idaction_url` INTEGER(11) UNSIGNED NULL  DEFAULT NULL... ^[[ADone. [1 / 1]

Piwik has been successfully updated!

PHP Fatal error:  Call to undefined function Piwik\posix_getgrgid() in /var/www/vhosts/***/httpdocs/core/Filechecks.php on line 268
server4:~$ php /var/www/vhosts/***/httpdocs/console core:update
```

Edit:
Found a forum post, but i'm not using XCache:
https://forum.piwik.org/t/fatal-errors-2-16-4-15-update-errors-xcache-issue/21225

I found out, that my php version is lower than recommended. Could this be the cause of this issue?

Please note, that my dashboard is working and tracking seems to be fine, too.
I just got this error during the update.

**Comment by pouyana** [comment]
Can confirm having the same issue on:
`PHP 5.5.14 (cli) 
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
    with Zend OPcache v7.0.4-dev, Copyright (c) 1999-2014, by Zend Technologies
`

**Comment by hansfn** [comment]
I'm running PHP 5.6.27 and just reproduced the problem upgrading to 2.17.0.

The problem is a missing Posix extension for PHP. In other words, not a Piwik code bug, but maybe a documentation bug? The Posix extension is not listed as required on http://piwik.org/docs/requirements/

**Comment by tsteur** [comment]
It is kind of a Piwik bug and I will provide a fix for Piwik 2.17.1 and 3.0.0

**Comment by mattab** [comment]
fixed in #10817 and #10825
